Mathlib/SetTheory/Ordinal/Principal.lean

```diff
@@ -195,65 +204,63 @@ theorem principal_add_iff_add_lt_ne_self : Principal (· + ·) a ↔ ∀ b < a,
     rcases exists_lt_add_of_not_principal_add ha with ⟨b, hb, c, hc, rfl⟩
     exact (H b hb c hc).irrefl⟩
 
+theorem principal_add_iff_add_self_lt : Principal (· + ·) a ↔ ∀ b < a, b + b < a :=
+  principal_iff_of_monotone
+    (fun x _ _ h => add_le_add_left h x) (fun x _ _ h => add_le_add_right h x)
+
 theorem principal_add_omega0 : Principal (· + ·) ω :=
-  principal_add_iff_add_left_eq_self.2 fun _ => add_omega0
-
-theorem add_omega0_opow (h : a < ω ^ b) : a + ω ^ b = ω ^ b := by
-  refine le_antisymm ?_ (le_add_left _ a)
-  induction' b using limitRecOn with b _ b l IH
-  · rw [opow_zero, ← succ_zero, lt_succ_iff, Ordinal.le_zero] at h
-    rw [h, zero_add]
-  · rw [opow_succ] at h
-    rcases (lt_mul_of_limit isLimit_omega0).1 h with ⟨x, xo, ax⟩
-    apply (add_le_add_right ax.le _).trans
-    rw [opow_succ, ← mul_add, add_omega0 xo]
-  · rcases (lt_opow_of_limit omega0_ne_zero l).1 h with ⟨x, xb, ax⟩
-    apply (((isNormal_add_right a).trans <| isNormal_opow one_lt_omega0).limit_le l).2
-    intro y yb
-    calc a + ω ^ y ≤ a + ω ^ max x y :=
-      add_le_add_left (opow_le_opow_right omega0_pos (le_max_right x y)) _
-    _ ≤ ω ^ max x y :=
-      IH _ (max_lt xb yb) <| ax.trans_le <| opow_le_opow_right omega0_pos <| le_max_left x y
-    _ ≤ ω ^ b :=
-      opow_le_opow_right omega0_pos <| (max_lt xb yb).le
+  principal_add_iff_add_left_eq_self.2 fun _ ↦ add_omega0
 
-@[deprecated (since := "2024-09-30")]
-alias add_omega_opow := add_omega0_opow
+-- `add_omega0` is proven in the Arithmetic file.
+
+theorem add_of_le_omega0 : a < ω → ω ≤ b → a + b = b :=
+  principal_add_omega0.add_absorp_of_le
 
-theorem principal_add_omega0_opow (o : Ordinal) : Principal (· + ·) (ω ^ o) :=
-  principal_add_iff_add_left_eq_self.2 fun _ => add_omega0_opow
+theorem principal_add_omega0_opow (o : Ordinal) : Principal (· + ·) (ω ^ o) := by
```

Comment:
Why has this proof got a lot longer?

Reply:
I switched the order of proofs around, proving this before proving `add_omega0_opow`. Note that the two proofs combined are shorter than they were before.